include/clang/AST/Expr.h

```diff
@@ -2798,7 +2798,9 @@ class BoundsExpr : public Expr {
 /// classes).
 class CastExpr : public Expr {
 private:
-  enum { OP, BOUNDS, END_EXPR = 2 };
+  // CASTBOUNDS - for expanded bounds of cast expression
+  // SUBEXPRBOUNDS - for bounds of subexpression
```

Comment:
Could a comment describing BOUNDS too?  It is the declared bounds of the result of the cast expression.